api/v1alpha1/ratelimitfilter_types.go

```diff
@@ -0,0 +1,138 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// RateLimitFilter allows the user to limit the number of incoming requests
+// to a predefined value based on attributes within the traffic flow.
+type RateLimitFilter struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of HTTPRoute.
+	Spec RateLimitFilterSpec `json:"spec"`
+}
```

Comment:
Has there been any discussion on status etc.?

What happens if the filter is configured incorrectly, how does a user get that feedback? Status on this resource or on the HTTPRoute?

Seems like at least mentioning some status condition that will be set on HTTPRoute is relevant here

Reply:
kept it out to limit the spec, will raise a issue post this PR merge to ensure status design is being tracked

Reply:
@arkodg can you update the PR based on the rate limiting infra management we discussed yesterday? Either add status and the details for exposing rate limit infra management through EnvoyGateway/EnvoyProxy APIs or create a tracker issue on this topic and link in the Design Decisions section of this PR.

Reply:
I would like to limit in the scope of this PR and and introduce the notion of the added infra in a future PR
The overall epic highlights the need for it https://github.com/envoyproxy/gateway/issues/670